### PR TITLE
WIP  Close of File Streams

### PR DESCRIPTION
**Pull request by rehammuzzamil**

### CODE REVIEW

File: src/main/java/org/opensrp/web/controller/UniqueIdController.java [hunk 1]
```diff
@@ -113,6 +117,22 @@
 			logger.error("", e);
 			message = "Sorry, an error occured when generating the qr code pdf";
 		}
+		finally {
```
Comment (maimoonak) [inline src/main/java/org/opensrp/web/controller/UniqueIdController.java:120]: can we use try with resources... https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/controller/UniqueIdController.java]: Yes, updated PR with try with resources.

File: src/main/java/org/opensrp/web/controller/UniqueIdController.java [hunk 2]
```diff
@@ -71,37 +71,33 @@
...
-				response.setHeader("Expires", "0");
-				response.setHeader("Cache-Control", "must-revalidate, post-check=0, pre-check=0");
-				response.setHeader("Pragma", "public");
-				response.setContentType("application/pdf");
-				response.setHeader("Content-Disposition", "attachment; filename=" + fileName);
-
-				OutputStream os = response.getOutputStream();
 				byteArrayOutputStream.writeTo(os);
 				os.flush();
 				os.close();
```
Comment (githengi) [inline src/main/java/org/opensrp/web/controller/UniqueIdController.java:103]: Remove this as it should be closed automatically

File: src/main/java/org/opensrp/web/controller/UniqueIdController.java [hunk 3]
```diff
@@ -71,37 +71,33 @@
...
-			SimpleDateFormat df = new SimpleDateFormat("dd-MM-yyyy-HHmmss");
-
-			String fileName = "QRCodes_".concat(df.format(new Date())).concat("_").concat(currentPrincipalName)
-					.concat("_" + numberToGenerate + ".pdf");
-			ByteArrayOutputStream byteArrayOutputStream = PdfUtil.generatePdf(idsToPrint, 140, 140, 1, 5);
 			if (byteArrayOutputStream.size() > 0) {
 				//mark ids as used
-				FileOutputStream fileOutputStream = new FileOutputStream(qrCodesDir + File.separator + fileName);
 				fileOutputStream.write(byteArrayOutputStream.toByteArray());
 				fileOutputStream.close();
```
Comment (githengi) [inline src/main/java/org/opensrp/web/controller/UniqueIdController.java:99]: Remove this as it should be closed automatically

File: src/main/java/org/opensrp/web/controller/UniqueIdController.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/controller/UniqueIdController.java:103]: Do I need to remove os.flush() as well, as the stream will be flushed automatically on close. Please confirm @githengi .

**Pull request merged**